### DISCUSSION TITLE
Question: When should a filesystem autoextend?

### DISCUSSION THREAD

**Discussion by ingothierack** [stratis-storage/project#406] [Q&A]
When should the filesystem extend?

I have a filesystem with comes to the limit of 1 TB
df tells me i'm on 95% of usage and 57 GB left.

stratis fs list
`stratis_pool01  fs_archiv        980.01 GiB   Mar 01 2019 13:10  /stratis/stratis_pool01/fs_archiv        45fa24827f264f24a61028fbb1d3824a
`
df
`/dev/mapper/stratis-1-210eb8dac6cb440993454dac6a885dae-thin-fs-45fa24827f264f24a61028fbb1d3824a 1073217536 1015409172   57808364  95% /mnt/archiv
`

Would it be if it goes to 100% or before?

I'm currently on latest which provided by fedory 31
stratisd-1.0.5-1.module_f31+6320+bf3c8975.x86_64
stratis-cli-1.0.4-1.module_f31+6320+bf3c8975.x86_64

**Comment by jbaublitz** [comment]
@ingothierack Hi! I've worked on this part of the code a fair amount for stratis-storage/stratisd#2814 and you may find some helpful information there.

I'm happy to try to help figure out your specific problem too, so could you give me the pool size and pool usage? To get a little bit ahead of myself, that version relies exclusively on the thin pool low water mark to check filesystem usage. This can be a problem because filesystems sometimes buffer a fair amount of data before flushing them to the layer below and I have observed cases where low water mark handling can get a bit delayed as a result. However, the low water mark for filesystems seems to be 8 GiB. Can you verify that less than 8 GiB on the filesystem causes it to be extended?
